internal/builtinfunctions/functions.go

```diff
@@ -0,0 +1,428 @@
+}
+
+// TODO: Webb can add a function called floatToStringAdvanced that allows other float formats
+
+func getBooleanToStringFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"boolToString",
+		[]schema.Type{schema.NewBoolSchema()},
+		schema.NewStringSchema(nil, nil, regexp.MustCompile(`^true|false$`)),
+		false,
+		schema.NewDisplayValue(
+			schema.PointerTo("boolToString"),
+			schema.PointerTo(
+				"Represents `true` as \"true\", and `false` as \"false\".",
+			),
+			nil,
+		),
+		strconv.FormatBool, // Wrap go standard lib function.
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getStringToIntFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"stringToInt",
+		[]schema.Type{schema.NewStringSchema(nil, nil, regexp.MustCompile(`^-?(?:0|[1-9]\d*)$`))},
+		schema.NewIntSchema(nil, nil, nil),
+		true,
+		schema.NewDisplayValue(
+			schema.PointerTo("stringToInt"),
+			schema.PointerTo(
+				"Interprets the string as a base-10 integer. Can fail if the input is not a valid integer.",
+			),
+			nil,
+		),
+		func(s string) (int64, error) {
+			return strconv.ParseInt(s, 10, 0)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getStringToFloatFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"stringToFloat",
+		[]schema.Type{schema.NewStringSchema(nil, nil, nil)},
+		schema.NewFloatSchema(nil, nil, nil),
+		true,
+		schema.NewDisplayValue(
+			schema.PointerTo("stringToFloat"),
+			schema.PointerTo(
+				"Converts the string s to a 64-bit floating-point number\n\n"+
+					"Accepts decimal and hexadecimal floating-point numbers\n"+
+					"as defined by the Go syntax for floating point literals\n"+
+					"https://go.dev/ref/spec#Floating-point_literals.\n"+
+					"If s is well-formed and near a valid floating-point number,\n"+
+					"ParseFloat returns the nearest floating-point number rounded\n"+
+					"using IEEE754 unbiased rounding.\n\n"+
+					"Returns NumError when an invalid input is received.",
+			),
+			nil,
+		),
+		func(s string) (float64, error) {
+			return strconv.ParseFloat(s, 64)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getStringToBoolFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"stringToBool",
+		[]schema.Type{
+			schema.NewStringSchema(
+				nil,
+				nil,
+				regexp.MustCompile(`[Tt]rue|TRUE|[Ff]alse|FALSE|[tTfF]|[01]$`),
+			)},
+		schema.NewBoolSchema(),
+		true,
+		schema.NewDisplayValue(
+			schema.PointerTo("stringToBool"),
+			schema.PointerTo(
+				"Interprets the input as a boolean.\n"+
+					"Accepts `1`, `t`, `T`, `true`, `TRUE`, and `True` for true.\n"+
+					"Accepts `0`, 'f', 'F', 'false', 'FALSE', and 'False' for false.\n"+
+					"Returns an error if any other value is input.",
+			),
+			nil,
+		),
+		strconv.ParseBool, // Wrap go standard lib function.
```

Comment:
I'm not quite sure what "enforce" and "proper" mean in this context, since the Go rules are rather specific and disjoint. We're not saying "oh, it has to be all lowercase" because Go enables some uppercase. Yeah, if a YAML workflow designer is motivated to "enforce" the arbitrary Go rules, we make it easy ... I just don't see any value in that, and calling it "proper" seems rather unnecessarily biased. 😁

Reply:
iNterEsting PerSPEctive.